src/taco-utils/commands.ts

```diff
@@ -50,14 +50,19 @@ module TacoUtility {
          */
         export interface ICommand {
             name: String;
-            run(data: ICommandData): Q.Promise<ICommandTelemetryProperties>;
+            run(args: string[]): Q.Promise<ICommandTelemetryProperties>;
             canHandleArgs(data: ICommandData): boolean;
         }
 
+        export interface ISubCommand {
```

Comment:
Should this interface extend `ICommand`? Slightly cleaner I think, that way you don't have to duplicate `name` and `canHandleArgs`, and it should still let you override `run`